KEP-4049: Add StorageCapacityScoring feature gate documents

cupnes opened the pull request:
### Description

This PR updates existing documentation to include the addition of the StorageCapacityScoring feature gate of KEP-4049  as an alpha version.

### Issue

- https://github.com/kubernetes/enhancements/issues/4049

sftim commented on content/en/docs/concepts/scheduling-eviction/scheduling-framework.md:
```diff
@@ -132,6 +132,9 @@ scheduler will call each scoring plugin for each node. There will be a well
 defined range of integers representing the minimum and maximum scores. After the
 [NormalizeScore](#normalize-scoring) phase, the scheduler will combine node
 scores from all plugins according to the configured plugin weights.
+With the enabling of the `StorageCapacityScoring` feature gate, nodes can be
+scored based on their available storage capacity. This only applies when using
+no static provisioning (dynamic provisioning only).
```
Try something like this:
```suggestion

#### Capacity scoring {#scoring-capacity}

{{< feature-state feature_gate_name="StorageCapacityScoring" >}}

As an alpha feature, nodes can be scored based on their available storage capacity.
This only applies when not using static provisioning
([dynamic provisioning](/docs/concepts/storage/dynamic-provisioning/) only).
```

See https://kubernetes.io/docs/contribute/new-content/new-features/#ready-for-review-feature-gates for more details about the extra metadata you need to create so that this will work.

cupnes replied:
Thank you for your suggestion. I am currently implementing the feature, and I will address it as soon as it is completed.

cupnes commented:
@sftim @ahg-g @salaxander @natalisucks 
I have addressed your comments and believe the work on the document is complete. Please review it. If anything is missing, let me know and I will fix it.

tengqm commented:
/lgtm

cupnes commented:
The alpha of KEP-4049 will skip version 1.32 ([Ref](https://github.com/kubernetes/enhancements/issues/4049#issuecomment-2461830519)). Therefore, I will close this PR that was opened against the dev-1.32 branch of the k/website repository.
